pkg/proxy/proxy_listener.go

```diff
@@ -109,6 +109,10 @@ func (l *Listener) Start() {
 
 	case "https":
 		proxy := l.NewHTTPProxy()
+
+		// Receive client certificate if provided
+		l.TLSConfig.ClientAuth = tls.RequestClientCert
```

Comment:
curl handled the certificate request well, even without client certificate. Not sure if any http client would.
Handshake reference:
https://golang.org/src/crypto/tls/handshake_server.go

Reply:
By setting this to tls.RequestClientCert, you change the default for all https requests regardless of LB pool, to request a client certificate.
the result is that for all requests, the browser must specificaly send a no certificate message with all requests, creating additional overhead.
this should be a configurable option passed in the config up on loading as part of the tls config provided via the listener

Reply:
Good point. I created a setting for it.